### DISCUSSION TITLE
Thoughts about http authentication?

### DISCUSSION THREAD

**Discussion by blacknell** [blakeblackshear/frigate#8565] [Ask A Question]
I realise that frigate has stayed away from providing a means of authentication to the front end. 

I'm doing this via apache reverse proxy and have done so on other systems without problem. 

This time though I'm getting multiple requests for auth after entering details. I don't know why but it seems the 2nd one is for /ws requests. 

Any ideas?

**Comment by hagak** [comment]
Not sure how you are doing the auth if it is completely within apache then it maybe difficult to resolve.  Or it could be as simple as making sure you are "protecting" a broad enough scope.

I would recommend using an OIDC provider like keycloak or authelia, this will provide a token that will be used on every request.  Also gives you a proper SSO tool that you can use with other applications.
